src/emcpy/plots/create_plot.py

```diff
@@ -0,0 +1,350 @@
+        Add data layer onto figure.
+
+        Args:
+            plot_list : (array-like) List of map plot objects from emcpy
+        """
+
+        plot_dict = {
+            'scatter': self._scatter,
+            'histogram': self._histogram,
+            'line_plot': self._lineplot
+        }
+
+        for obj in plot_list:
+            try:
+                plot_dict[obj.plottype](obj)
+            except KeyError:
+                raise TypeError(f'{obj} is not a valid plot type.' +
+                                'Current plot types supported are:\n' +
+                                f'{" | ".join(feature_dict.keys())}"')
+
+    def _density_scatter(self, plotobj):
+        """
+        Uses Scatter Object to plot density scatter colored by
+        2d histogram.
+        """
+        data, x_e, y_e = np.histogram2d(plotobj.x, plotobj.y,
+                                        bins=plotobj.bins,
+                                        density=True)
+        z = interpn((0.5*(x_e[1:] + x_e[:-1]), 0.5*(y_e[1:]+y_e[:-1])),
+                    data, np.vstack([plotobj.x, plotobj.y]).T,
+                    method="splinef2d", bounds_error=False)
+        # To be sure to plot all data
+        z[np.where(np.isnan(z))] = 0.0
+        # Sort the points by density, so that the densest
+        # points are plotted last
+        if plotobj.sort:
+            idx = z.argsort()
+            x, y, z = plotobj.x[idx], plotobj.y[idx], z[idx]
+        cs = self.ax.scatter(x, y, c=z,
+                             s=plotobj.markersize,
+                             cmap=plotobj.cmap,
+                             label=plotobj.label)
+        norm = Normalize(vmin=np.min(z), vmax=np.max(z))
+
+        if plotobj.colorbar:
+            self.cs = cs
+
+    def _scatter(self, plotobj):
+        """
+        Uses Scatter object to plot on axis.
+        """
+        self.fig = plt.figure(figsize=self.figsize)
+        self.ax = self.fig.add_subplot(111)
+
+        # checks to see if density has been added as an attribute
+        # to the plotting object and if it's True
+        if 'density' in dir(plotobj) and plotobj.density:
+            self._density_scatter(plotobj)
+
+        else:
+            s = self.ax.scatter(plotobj.x, plotobj.y,
+                                s=plotobj.markersize,
+                                color=plotobj.color,
+                                marker=plotobj.marker,
+                                vmin=plotobj.vmin,
+                                vmax=plotobj.vmax,
+                                alpha=plotobj.alpha,
+                                linewidths=plotobj.linewidths,
+                                edgecolors=plotobj.edgecolors,
+                                label=plotobj.label)
+
+        # checks to see if linear regression has been added as an attribute
+        # to the plotting object and if it's True
+        if 'linear_regression' in dir(plotobj) and plotobj.linear_regression:
```

Comment:
Same question as above, I guess the question is, is just the second part (after the and) sufficient?